sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -494,7 +495,19 @@ protected final void removeTemporaryFiles(List<String> knownFiles, PipelineOptio
       LOG.debug("Removing temporary bundle output files in {}.", tempDirectory);
       FileOperations fileOperations =
           FileOperationsFactory.getFileOperations(tempDirectory, options);
-      fileOperations.removeDirectoryAndFiles(tempDirectory, knownFiles);
+      IOChannelFactory factory = IOChannelUtils.getFactory(tempDirectory);
+      Collection<String> matches = factory.match(factory.resolve(tempDirectory, "*"));
```

Comment:
@peihe, resolving  tempDirectory with asterisks  [doesn't work on windows](http://stackoverflow.com/questions/27522581/asterisks-in-java-path).  It would be great if DirectRunner and UT can run on Windows.